src/MongoDB.Driver.Core/Core/Connections/ConnectionInitializer.cs

```diff
@@ -92,59 +107,124 @@ public ConnectionDescription InitializeConnection(IConnection connection, Cancel
 
             return description;
         }
+        
+        internal BsonDocument AddAuthenticatorCustomizationsToIsMasterCommand(
+            BsonDocument command,
+            IReadOnlyList<IAuthenticator> authenticators)
+        {
```

Comment:
Let's make up our minds on whether these methods mutate the document or not. Let's either:

1. Declare void and mutate the document
2. Declare returns BsonDocument and return a new document leaving the original unchanged

I vote for 1 since this is all internal and noone else ever sees this command. It's created once, executed, and thrown away.

Reply:
The current methods do mutate the document: they're fluent interfaces for convenience of  chaining method calls. I could make them void if you prefer that style at the price of a few extra lines of code.

Reply:
Left as fluent interfaces, as discussed offline.